### ISSUE TITLE
Definition List compounds bottom margin with each nested level

### ISSUE THREAD

**Issue by polarathene** [squidfunk/mkdocs-material#2501]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

With each nested level of definition list, the bottom margin seems to expand.

This appears to be due to `display: flow-root` (which the expected behaviour shows when removed). I'm not entirely sure why this rule is used to [generate new block formatting context](https://developer.mozilla.org/en-US/docs/Web/CSS/display), so it's probably not a good solution (unless it were changed to not apply to nested children?).

The behaviour examples shown below is with excessive nesting. My actual use case is not as bad, but noticeable enough that I figured I should raise an issue about it.

![Screenshot_20210329_154955](https://user-images.githubusercontent.com/5098581/112780973-706b8e80-90a6-11eb-90a5-451d8cb2486d.png)

```markdown
???+ info "Syntax"

    1. Declaration: `=== [tab-title]`

        `[tab-title]`
        
        : Provide a quoted string as the name of the tab. Markdown content is supported.

    2. Tab content is provided from the next line onward, assigned by indentation of 4 spaces.
    3. There should be a blank line before and after the tab content.
    4. Repeat steps for grouping additional content under related tabs.
```

### Expected behavior

![Screenshot_20210329_153248](https://user-images.githubusercontent.com/5098581/112779836-110c7f00-90a4-11eb-944e-5186bd72b152.png)

### Actual behavior

![Screenshot_20210329_153025](https://user-images.githubusercontent.com/5098581/112779777-f63a0a80-90a3-11eb-8639-ee226b66b0a4.png)

### Steps to reproduce the bug

1. Enable and use the `def_list` markdown extension in `mkdocs.yml`.
2. Add the markdown below to a doc page and build/preview the rendered output:

    ```markdown
    a
    
    : b
    
        : c
    
            : d
    
                : e
    
    f
    ```

### Package versions

Current release of the `mkdocs-material` Docker container.

### Project configuration

Standard instance should do the trick, this is affected by `mkdocs-material` CSS.

### System information

Manjaro Linux with Firefox.

**Comment by squidfunk** [comment]
Thanks for reporting. That shouldn't happen. I've added `display: flow-root`, so those elements won't bleed in to inline admonitions, but it appears that it's not necessary. I've removed it in c011da1fa. Let's see if that breaks something else.

**Comment by squidfunk** [comment]
Released as part of 7.1.1.
